java/vector/src/main/java/org/apache/arrow/vector/ipc/message/ArrowRecordBatch.java

```diff
@@ -194,12 +190,17 @@ public int writeTo(FlatBufferBuilder builder) {
     int nodesOffset = FBSerializables.writeAllStructsToVector(builder, nodes);
     RecordBatch.startBuffersVector(builder, buffers.size());
     int buffersOffset = FBSerializables.writeAllStructsToVector(builder, buffersLayout);
-    int compressOffset = bodyCompression.writeTo(builder);
+    int compressOffset = 0;
+    if (bodyCompression != null) {
```

Comment:
I would think the only change really necessary here would have been to compare bodyCompression == NO_COMPRESSION_CODEC that way you can keep things non-null everyplace else within the code?

Reply:
@emkornfield Thanks for your good idea. I have updated the code accordingly.